nestjs/src/courses/task-verifications/self-education.service.ts

```diff
@@ -0,0 +1,147 @@
+import { CourseTask } from '@entities/courseTask';
+import { TaskVerification } from '@entities/taskVerification';
+import { ForbiddenException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { WriteScoreService } from '../score';
+
+type CorrectAnswer = {
+  index: number;
+  value: (number | number[])[];
+  isCorrect: boolean;
+};
+
+type SelfEducationVerificationParams = {
+  courseId: number;
+  courseTask: CourseTask;
+  studentId: number;
+  studentAnswers: {
+    index: number;
+    value: (number | number[])[];
+  }[];
+};
+
+type SelfEducationAttributes = {
+  public: {
+    maxAttemptsNumber: number;
+    numberOfQuestions: number;
+    tresholdPercentage: number;
+    strictAttemptsMode?: boolean;
+    oneAttemptPerNumberOfHours?: number;
+    questions: {
+      question: string;
+      answers: string[];
+      multiple: boolean;
+    }[];
+  };
+  answers: (number | number[])[];
+};
+
+@Injectable()
+export class SelfEducationService {
+  constructor(
+    @InjectRepository(TaskVerification)
+    private readonly taskVerificationsRepository: Repository<TaskVerification>,
+
+    private readonly writeScoreService: WriteScoreService,
+  ) {}
+
+  public async createSelfeducationVerification({
```

Comment:
This method is too big, maybe split it a little bit? :)

Reply:
it's only about 70 lines and pretty straightforward